internal/provider/resource_directory.go

```diff
@@ -368,3 +379,25 @@ func (rs *directoryResource) Delete(ctx context.Context, req resource.DeleteRequ
 func (rs *directoryResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
 }
+
+func sortDiretoryFeatures(directoryFeatures []string) []string {
+
+	//Directory Features must be handed to the CLI in a well defined order.
+	//In case Terraform sorts the entries alphabetically or they are handed over in the wrong sequence, we make sure
+	//that they are handed over correctly
+	directoryFeaturesSorted := []string{}
+
+	if slices.Contains(directoryFeatures, "DEFAULT") {
+		directoryFeaturesSorted = append(directoryFeaturesSorted, "DEFAULT")
+	}
+
+	if slices.Contains(directoryFeatures, "ENTITLEMENTS") {
+		directoryFeaturesSorted = append(directoryFeaturesSorted, "ENTITLEMENTS")
+	}
+
+	if slices.Contains(directoryFeatures, "AUTHORIZATIONS") {
+		directoryFeaturesSorted = append(directoryFeaturesSorted, "AUTHORIZATIONS")
+	}
```

Comment:
Could you please refactor this part and make the strings constant?

Reply:
Switched to constants